sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py

```diff
@@ -0,0 +1,155 @@
+            :dedent: 4
+            :caption: Create an InteractiveBrowserCredential.
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self._allow_broker = kwargs.pop("allow_broker", None)
+        self._parent_window_handle = kwargs.pop("parent_window_handle", None)
+        self._enable_msa_passthrough = kwargs.pop("enable_msa_passthrough", False)
+        redirect_uri = kwargs.pop("redirect_uri", None)
+        if redirect_uri:
+            self._parsed_url = urlparse(redirect_uri)
+            if not (self._parsed_url.hostname and self._parsed_url.port):
+                raise ValueError('"redirect_uri" must be a URL with port number, for example "http://localhost:8400"')
+        else:
+            self._parsed_url = None
+
+        self._login_hint = kwargs.pop("login_hint", None)
+        self._timeout = kwargs.pop("timeout", 300)
+        client_id = kwargs.pop("client_id", DEVELOPER_SIGN_ON_CLIENT_ID)
+        super(InteractiveBrowserBrokerCredential, self).__init__(client_id=client_id, **kwargs)
+
+    @wrap_exceptions
+    def _request_token(self, *scopes: str, **kwargs: Any) -> Dict:
+        scopes = list(scopes)  # type: ignore
+        claims = kwargs.get("claims")
+        app = self._get_app(**kwargs)
+        port = self._parsed_url.port if self._parsed_url else None
+
+        try:
+            result = app.acquire_token_interactive(
+                scopes=scopes,
+                login_hint=self._login_hint,
+                claims_challenge=claims,
+                timeout=self._timeout,
+                prompt="select_account",
+                port=port,
+                parent_window_handle=self._parent_window_handle,
+                enable_msa_passthrough=self._enable_msa_passthrough,
+            )
+        except socket.error as ex:
+            raise CredentialUnavailableError(message="Couldn't start an HTTP server.") from ex
+        if "access_token" not in result and "error_description" in result:
+            if within_dac.get():
+                raise CredentialUnavailableError(message=result["error_description"])
+            raise ClientAuthenticationError(message=result.get("error_description"))
+        if "access_token" not in result:
+            if within_dac.get():
+                raise CredentialUnavailableError(message="Failed to authenticate user")
+            raise ClientAuthenticationError(message="Failed to authenticate user")
+
+        # base class will raise for other errors
+        return result
+
+    def _get_app(self, **kwargs: Any) -> msal.ClientApplication:
+        tenant_id = resolve_tenant(
+            self._tenant_id, additionally_allowed_tenants=self._additionally_allowed_tenants, **kwargs
+        )
+
+        client_applications_map = self._client_applications
+        capabilities = None
+        token_cache = self._cache
+
+        app_class = msal.ConfidentialClientApplication if self._client_credential else msal.PublicClientApplication
```

Comment:
Can we remove this line and use `msal.PublicClientApplication directly? Don't think we'd ever use ConfidentialClientApplication here, right?

Reply:
I thought that before but I don't think we want to make it different from its base IntBrowserCred.